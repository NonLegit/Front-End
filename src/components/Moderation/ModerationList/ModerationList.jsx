/* eslint-disable react/no-array-index-key */
import {
  List, ListItemButton, ListItemText, ListItemIcon,
  Collapse,
} from '@mui/material';
import { useParams, useNavigate } from 'react-router-dom';
import ExpandMore from '@mui/icons-material/ExpandMore';
import * as React from 'react';
import BackupTableOutlinedIcon from '@mui/icons-material/BackupTableOutlined';
import PersonOutlineOutlinedIcon from '@mui/icons-material/PersonOutline';
import SellOutlinedIcon from '@mui/icons-material/SellOutlined';
import ArticleOutlinedIcon from '@mui/icons-material/ArticleOutlined';
import FormatListBulletedOutlinedIcon from '@mui/icons-material/FormatListBulletedOutlined';
import SettingsOutlinedIcon from '@mui/icons-material/SettingsOutlined';
import SignalCellularAltOutlinedIcon from '@mui/icons-material/SignalCellularAltOutlined';
import StyledList, { HeaderText, HeaderLink, HeaderSubpage } from './styles';

function ModerationList() {
  const { subReddit, subTitle } = useParams();

  const [openModList, setOpenModList] = React.useState(0);
  const handleClickModList = () => {
    setOpenModList(!openModList);
  };
  const navigate = useNavigate();

  // navigate
  const handleClick = (e, subPage) => {
    navigate(`${subPage}`);
  };

  return (
    <StyledList>
      <ListItemButton disableTouchRipple onClick={handleClickModList}>
        {/* <ListItemText primary="Mod" sx={{ color: 'black', paddingLeft: '5px' }} /> */}
        <HeaderText>
          <HeaderLink underline="none">
            r/
            {subReddit}
          </HeaderLink>
          <HeaderSubpage>
          &nbsp;/&nbsp;
            {subTitle}
          </HeaderSubpage>
        </HeaderText>
        <ExpandMore sx={{ color: '#757575', fontSize: 20 }} />
      </ListItemButton>
      <Collapse
        in={Boolean(openModList)}
        sx={{
          position: 'absolute', width: '270px', left: '-50px', top: '45px',
        }}
        timeout="auto"
        unmountOnExit
      >
        <List component="div" disablePadding>
          <ListItemButton disabled>
            <ListItemIcon><BackupTableOutlinedIcon /></ListItemIcon>
            <ListItemText primary="QUEUES" sx={{ color: 'black', paddingLeft: '5px' }} />
          </ListItemButton>
          {['Spam', 'Edited', 'Unmoderated'].map((items, index) => (
            <ListItemButton key={`${index + 0}`} onClick={(e) => { handleClick(e, `${items}`); }}>
              <ListItemText primary={items} />
            </ListItemButton>
          ))}
          <ListItemButton disabled>
            <ListItemIcon><PersonOutlineOutlinedIcon /></ListItemIcon>
            <ListItemText primary="USER MANAGEMENT" sx={{ color: 'black', paddingLeft: '5px' }} />
          </ListItemButton>
          {['Banned', 'Muted', 'Approved', 'Moderators'].map((items, index) => (
            <ListItemButton key={`${index + 0}`} onClick={(e) => { handleClick(e, `${items}`); }}>
              <ListItemText primary={items} />
            </ListItemButton>
          ))}
          <ListItemButton disabled>
            <ListItemIcon><SellOutlinedIcon /></ListItemIcon>
            <ListItemText primary="FLAIR & EMOJIES" sx={{ color: 'black', paddingLeft: '5px' }} />
          </ListItemButton>
          {['Post flair'].map((items, index) => (
            <ListItemButton key={`${index + 0}`} onClick={(e) => { handleClick(e, `${items}`); }}>
              <ListItemText primary={items} />
            </ListItemButton>
          ))}
          <ListItemButton disabled>
            <ListItemIcon><ArticleOutlinedIcon /></ListItemIcon>
            <ListItemText primary="RULES AND REGULATIONS" sx={{ color: 'black', paddingLeft: '5px' }} />
          </ListItemButton>
          {['Rules', 'Removal reasons', 'Content controls'].map((items, index) => (
            <ListItemButton key={`${index + 0}`} onClick={(e) => { handleClick(e, `${items}`); }}>
              <ListItemText primary={items} />
            </ListItemButton>
          ))}
          <ListItemButton disabled>
            <ListItemIcon><FormatListBulletedOutlinedIcon /></ListItemIcon>
            <ListItemText primary="CONTENT" sx={{ color: 'black', paddingLeft: '5px' }} />
          </ListItemButton>
          {['Scheduled posts'].map((items, index) => (
            <ListItemButton key={`${index + 0}`} onClick={() => { handleClick(`${items}`); }}>
              <ListItemText primary={items} />
            </ListItemButton>
          ))}
          <ListItemButton disabled>
            <ListItemIcon><SettingsOutlinedIcon /></ListItemIcon>
            <ListItemText primary="OTHER" sx={{ color: 'black', paddingLeft: '5px' }} />
          </ListItemButton>
          {['Comunity settings', 'Comunity apperance'].map((items, index) => (
            <ListItemButton key={`${index + 0}`} onClick={() => { handleClick(`${items}`); }}>
              <ListItemText primary={items} />
            </ListItemButton>
          ))}
          <ListItemButton disabled>
            <ListItemIcon><SignalCellularAltOutlinedIcon /></ListItemIcon>
            <ListItemText primary="COMMUNITY ACTIVITY" sx={{ color: 'black', paddingLeft: '5px' }} />
          </ListItemButton>
          {['Traffic stats'].map((items, index) => (
            <ListItemButton key={`${index + 0}`} onClick={() => { handleClick(`${items}`); }}>
              <ListItemText primary={items} />
            </ListItemButton>
          ))}
        </List>
      </Collapse>
    </StyledList>
  );
}

export default ModerationList;
