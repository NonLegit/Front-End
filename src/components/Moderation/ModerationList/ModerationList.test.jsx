import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import ModerationList from './ModerationList';

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

test('test snapshot', async () => {
  const tree = renderer.create(<ModerationList />).toJSON();
  expect(tree).toMatchSnapshot();
});
