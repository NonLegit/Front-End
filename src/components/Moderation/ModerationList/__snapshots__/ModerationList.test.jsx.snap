// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding css-1g93nk2-MuiList-root"
>
  <div
    className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="css-aw179r"
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1r83qox-MuiTypography-root-MuiLink-root"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        r/
      </a>
      <p
        className="MuiTypography-root MuiTypography-body1 css-dzy1yv-MuiTypography-root"
      >
         / 
      </p>
    </div>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-yoveok-MuiSvgIcon-root"
      data-testid="ExpandMoreIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
      />
    </svg>
  </div>
</ul>
`;
